src/platform/datapath_raw_xdp_win.c

```diff
@@ -1700,6 +1700,7 @@ CxPlatXdpTx(
     if (ProdCount > 0 ||
         (CompCount > 0 && XskRingProducerReserve(&Queue->TxRing, MAXUINT32, &TxIndex) != Queue->TxRing.Size)) {
         XskRingProducerSubmit(&Queue->TxRing, ProdCount);
```

Comment:
yeah, XskRingProducerSubmit could be conditional. the barrier does need to be part of the complete conditional, though, because the `(CompCount > 0 && XskRingProducerReserve(&Queue->TxRing, MAXUINT32, &TxIndex) != Queue->TxRing.Size)` case can also require a poke.

Reply:
`(XskRingProducerSubmit(&Queue->TxRing, ProdCount), TRUE))` I have never seen such an expression in C :) but I think I understand how it'd behave :)